lib/optimizely.rb

```diff
@@ -388,39 +388,37 @@ def get_feature_variable_for_type(feature_flag_key, variable_key, variable_type,
 
       variable_value = nil
       variable = @config.get_feature_variable(feature_flag, variable_key)
-      unless variable.nil?
-        variable_value = variable['defaultValue']
 
+      # Error message logged in ProjectConfig- get_feature_flag_from_key
+      return nil if variable.nil?
+
+      # Returns nil if type differs
+      if variable['type'] != variable_type
+        @logger.log(Logger::WARN,
+          "Requested variable type '#{variable_type}' but variable '#{variable_key}' is of type '#{variable['type']}'.")
+        return nil
+      else
         decision = @decision_service.get_variation_for_feature(feature_flag, user_id, attributes)
-        unless decision
-          @logger.log(Logger::INFO,
-            "User '#{user_id}' was not bucketed into any variation for feature flag '#{feature_flag_key}'. Returning the default variable value '#{variable_value}'.")
-        else
+        variable_value = variable['defaultValue']
+        if decision
           variation = decision['variation']
           variation_variable_usages = @config.variation_id_to_variable_usage_map[variation['id']]
           variable_id = variable['id']
-          unless variation_variable_usages and variation_variable_usages.key?(variable_id)
-            variation_key = variation['key']
-            @logger.log(Logger::DEBUG,
-              "Variable '#{variable_key}' is not used in variation '#{variation_key}'. Returning the default variable value '#{variable_value}'."
-            )
-          else
+          if variation_variable_usages and variation_variable_usages.key?(variable_id)
             variable_value = variation_variable_usages[variable_id]['value']
             @logger.log(Logger::INFO,
               "Got variable value '#{variable_value}' for variable '#{variable_key}' of feature flag '#{feature_flag_key}'.")
+          else
+            @logger.log(Logger::DEBUG,
+              "Variable '#{variable_key}' is not used in variation '#{variation['key']}'. Returning the default variable value '#{variable_value}'.")
           end
+        else
+          @logger.log(Logger::INFO,
+            "User '#{user_id}' was not bucketed into any variation for feature flag '#{feature_flag_key}'. Returning the default variable value '#{variable_value}'.")
         end
       end
 
-      unless variable_value.nil?
-        actual_variable_type = variable['type']
-        unless variable_type == actual_variable_type
-          @logger.log(Logger::WARN,
-            "Requested variable type '#{variable_type}' but variable '#{variable_key}' is of type '#{actual_variable_type}'.")
-        end
-
-        variable_value = Helpers::VariableType.cast_value_to_type(variable_value, variable_type, @logger)
-      end
+      variable_value = Helpers::VariableType.cast_value_to_type(variable_value, variable_type, @logger) if variable_value.present?
```

Comment:
no need of if. variable_value will be present in all cases

Reply:
Done!